src/server/game/Config/GameConfig.cpp

```diff
@@ -0,0 +1,210 @@
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "DatabaseEnv.h"
+#include "GameConfig.h"
+#include "Log.h"
+#include "Timer.h"
+#include "Util.h"
+
+GameConfig* GameConfig::instance()
+{
+    static GameConfig instance;
+    return &instance;
+}
+
+void GameConfig::CleanAll()
+{
+    _boolOptions.clear();
+    _intOptions.clear();
+    _floatOptions.clear();
+    _stringOptions.clear();
+}
+
+void GameConfig::AddBoolOption(std::string const& optionName, bool value /*= false*/)
+{
+    auto const& itr = _boolOptions.find(optionName);
+    if (itr != _boolOptions.end())
+    {
+        TC_LOG_FATAL("server.loading", "> Bool option (%s) is exist!", optionName.c_str());
+        return;
+    }
+
+    _boolOptions.insert(std::make_pair(optionName, value));
+}
+
+void GameConfig::AddIntOption(std::string const& optionName, int32 value /*= 0*/)
+{
+    auto const& itr = _intOptions.find(optionName);
+    if (itr != _intOptions.end())
+    {
+        TC_LOG_FATAL("server.loading", "> Int option (%s) is exist!", optionName.c_str());
+        return;
+    }
+
+    _intOptions.insert(std::make_pair(optionName, value));
+}
+
+void GameConfig::AddFloatOption(std::string const& optionName, float value /*= 1.0f*/)
+{
+    auto const& itr = _floatOptions.find(optionName);
+    if (itr != _floatOptions.end())
+    {
+        TC_LOG_FATAL("server.loading", "> Float option (%s) is exist!", optionName.c_str());
+        return;
+    }
+
+    _floatOptions.insert(std::make_pair(optionName, value));
+}
+
+void GameConfig::AddStringOption(std::string const& optionName, std::string const& value /*= ""*/)
+{
+    auto const& itr = _stringOptions.find(optionName);
+    if (itr != _stringOptions.end())
+    {
+        TC_LOG_FATAL("server.loading", "> Int option (%s) is exist!", optionName.c_str());
+        return;
+    }
+
+    _stringOptions.insert(std::make_pair(optionName, value));
+}
+
+void GameConfig::AddOption(std::string const& optionName, GameConfigType type, std::string const& value, std::string const& defaultValue)
+{
+    switch (type)
+    {
+    case GameConfigType::GAME_CONFIG_TYPE_BOOL:
+        AddBoolOption(optionName, value.empty() ? StringToBool(defaultValue) : StringToBool(value));
+        break;
+    case GameConfigType::GAME_CONFIG_TYPE_INT:
+        AddIntOption(optionName, value.empty() ? std::stoi(defaultValue) : std::stoi(value));
```

Comment:
please check if some recent changes in 3.3.5 branch added new features to transform a string to a int/float

Reply:
Yes yes, okay

Reply:
Hm ho to use `Trinity::StringTo` for `float` or this type doesn't support?

Reply:
float support soon, give me a day or two, i wanted to merge #25323

Reply:
Very good, i wait

Reply:
#25364